### PR TITLE
Removing unnecessary parts in scrappie to get nanopolish working on ARM

### PR DESCRIPTION
**Pull request by hasindu2008**
The extracted code from Scrappie contains certain parts with Intel SSE instructions. However, nanopolish does not use those parts at all. Hence, they were removed. Now Nanopolish should compile in non-Intel systems.

### CODE REVIEW

**Comment by junaruga** [comment]
I found 2 CI sites supporting ARM.

Ci: Work on ARM:
https://www.worksonarm.com/cluster/

Shippable
http://www.shippable.com
http://blog.shippable.com/shippable-arm-packet-deliver-native-ci-cd-for-arm-architecture

I think that we can consider to start to use the CI to test `nanopolish` with this PR's patch.

**Comment by junaruga** [comment]
I opened the ticket for the proposal to start ARM test on CI.
https://github.com/jts/nanopolish/issues/473

**Comment by junaruga** [comment]
@hasindu2008 could you tell us the output of `uname -a` to know your environment?

**Comment by hasindu2008** [comment]
@junaruga I have been running nanopolish on a few ARM systems

rock64 : Linux rock2 4.4.126-rockchip-ayufan-239 #1 SMP Sun May 27 18:38:24 UTC 2018 aarch64 aarch64 aarch64 GNU/Linux
Jetson : Linux jetson 4.4.38-tegra #1 SMP PREEMPT Thu May 17 00:15:19 PDT 2018 aarch64 aarch64 aarch64 GNU/Linux
Odroid XU4 : Linux odroidM 4.14.32-126 #1 SMP PREEMPT Wed Apr 4 03:59:40 UTC 2018 armv7l armv7l armv7l GNU/Linux

**Comment by hasindu2008** [comment]
@junaruga I have been running nanopolish on a few ARM systems

rock64 : Linux rock2 4.4.126-rockchip-ayufan-239 #1 SMP Sun May 27 18:38:24 UTC 2018 aarch64 aarch64 aarch64 GNU/Linux
Jetson : Linux jetson 4.4.38-tegra #1 SMP PREEMPT Thu May 17 00:15:19 PDT 2018 aarch64 aarch64 aarch64 GNU/Linux
Odroid XU4 : Linux odroidM 4.14.32-126 #1 SMP PREEMPT Wed Apr 4 03:59:40 UTC 2018 armv7l armv7l armv7l GNU/Linux

**Comment by hasindu2008** [comment]
@junaruga I have been running nanopolish on a few ARM systems

rock64 : Linux rock2 4.4.126-rockchip-ayufan-239 #1 SMP Sun May 27 18:38:24 UTC 2018 aarch64 aarch64 aarch64 GNU/Linux
Jetson : Linux jetson 4.4.38-tegra #1 SMP PREEMPT Thu May 17 00:15:19 PDT 2018 aarch64 aarch64 aarch64 GNU/Linux
Odroid XU4 : Linux odroidM 4.14.32-126 #1 SMP PREEMPT Wed Apr 4 03:59:40 UTC 2018 armv7l armv7l armv7l GNU/Linux

**Comment by junaruga** [comment]
@hasindu2008 alright. You are using both `aarch64` (ARM 64-bit) and `armv7l` (ARM 32-bit).

File: Makefile [hunk 1]
```diff
@@ -10,8 +10,8 @@ SUBDIRS := src src/hmm src/thirdparty src/thirdparty/scrappie src/common src/ali
 #Basic flags every build needs
 LIBS=-lz
 CXXFLAGS ?= -g -O3
-CXXFLAGS += -std=c++11 -fopenmp -fsigned-char
-CFLAGS ?= -O3 -std=c99
+CXXFLAGS += -std=c++11 -fopenmp -fsigned-char -D_FILE_OFFSET_BITS=64 #D_FILE_OFFSET_BITS=64 makes nanopolish work in 32 bit systems
+CFLAGS ?= -O3 -std=c99 -fsigned-char -D_FILE_OFFSET_BITS=64
```
Comment (junaruga) [inline Makefile]: This line has a tailing space. It can be removed.

File: Makefile [hunk 2]
```diff
@@ -10,8 +10,8 @@ SUBDIRS := src src/hmm src/thirdparty src/thirdparty/scrappie src/common src/ali
 #Basic flags every build needs
 LIBS=-lz
 CXXFLAGS ?= -g -O3
-CXXFLAGS += -std=c++11 -fopenmp -fsigned-char
-CFLAGS ?= -O3 -std=c99
+CXXFLAGS += -std=c++11 -fopenmp -fsigned-char -D_FILE_OFFSET_BITS=64 #D_FILE_OFFSET_BITS=64 makes nanopolish work in 32 bit systems
```
Comment (junaruga) [inline Makefile]: Seeing other comment parts, I think that below style looks better.

```
# D_FILE_OFFSET_BITS=64 makes nanopolish work in 32 bit systems
CXXFLAGS += -std=c++11 -fopenmp -fsigned-char -D_FILE_OFFSET_BITS=64
```

**Comment by junaruga** [comment]
Though I am not a project maintainer, I just commented for your code.
After #492 , you can rebase on the latest master branch, and verify it.
Currently there are 4 commits in this PR.
2nd, 3rd commit hash is merging commits. You can remove the commits when rebasing.

**Comment by jts** [comment]
Hi @hasindu2008 and @junaruga,

Sorry for taking so long but I finally merged this PR. Thanks again for the contribution.

Jared

**Pull request merged**